flow/middleware/tracing.go

```diff
@@ -0,0 +1,23 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/PeerDB-io/peerdb/flow/shared"
+)
+
+func RequestIdMiddleware() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (any, error) {
+		md, _ := metadata.FromIncomingContext(ctx)
+		requestIds := md.Get(shared.RequestIdKey.String())
+		requestId := uuid.NewString()
+		if requestIds != nil {
+			requestId = requestIds[0]
+		}
```

Comment:
```suggestion
		var requestId string
		if len(requestIds) > 1 {
			requestId = requestIds[0]
		} else {
			requestId := uuid.NewString()
		}
```
avoids panic on empty slice, avoids generating random uuid when requestIds not empty